packages/data-stores/src/queries/use-jetpack-search-ai.ts

```diff
@@ -0,0 +1,32 @@
+import { useQuery } from 'react-query';
+import wpcomRequest from 'wpcom-proxy-request';
+
+type AIResponseURL = {
+	url: string;
+	title: string;
+};
+
+export type JetpackSearchAIResult = {
+	response: string;
+	step: string;
+	urls: AIResponseURL[];
+	terms: string[];
+};
+
+export function useJetpackSearchAIQuery( siteId: number | string, query: string, stopAt: string ) {
+	return useQuery< JetpackSearchAIResult >(
+		[ query ],
+		async () =>
+			await wpcomRequest( {
+				path: `sites/${ siteId }/jetpack-search/ai/search`,
+				apiNamespace: 'wpcom/v2/',
+				apiVersion: '2',
+				query: `query=${ encodeURIComponent( query ) }&stop_at=${ stopAt }`,
+			} ),
+		{
+			refetchOnWindowFocus: false,
+			keepPreviousData: false,
```

Comment:
Why not? This query is expensive, if someone searches for "A", then "B", then "A", I think they should get a cached responsive for "A".

Reply:
The endpoint is a bit peculiar:

1. It caches by itself, quite heavily
2. Each new query is "moving the steps" a little further. In order to reduce lag, we can get early results by querying the endpoint with different ?stop_at until we have a result.

In the light of this I decided against multiple layers of caching

Reply:
Thanks for the context, really appreciated.